numpyro/infer/elbo.py

```diff
@@ -67,6 +67,14 @@ def loss_with_mutable_state(
         """
         raise NotImplementedError("This ELBO objective does not support mutable state.")
 
+    def can_infer_discrete(self):
```

Comment:
why not a property instead of a method?
```py
can_infer_discrete = False
```

Reply:
That works. I can also shorten the name to something that works a bit better in 'not' situations. Maybe `supports_discrete` ?

Reply:
maybe `has_discrete_support`? I like `can_infer_discrete` though.